### PR TITLE
Allow URL customisations for Google, Outlook and Yahoo

### PR DESCRIPTION
**Pull request by alies-dev**
Google, Outlook and Yahoo have custom GET parameters that it's currently not possible to set from Link class. This PR allow to add such parameters.

### CODE REVIEW

**Comment by alies-dev** [comment]
Hey @atymic. What do you think about this feature and the implementation?

File: tests/Generators/GoogleGeneratorTest.php [hunk 1]
```diff
@@ -35,4 +35,12 @@ public function it_correctly_generates_all_day_events_by_dates(): void
             $this->generator()->generate($this->createEventMultipleDaysViaStartEndWithTimezoneLink())
         );
     }
+
+    /** @test */
+    public function it_can_generate_an_url_with_custom_parameters(): void
+    {
+        $link = $this->createShortEventLink();
+
+        $this->assertMatchesSnapshot($link->google(['recur' => 'RRULE:FREQ=DAILY']));
```
Comment (atymic) [inline tests/Generators/GoogleGeneratorTest.php:44]: Oh google does support re-occuring!
Did you find documentation for the params?

**Review by atymic — COMMENTED** [review]
Looks great! Thanks @alies-dev

File: tests/Generators/GoogleGeneratorTest.php (hunk 1 above)
Comment (alies-dev) [reply tests/Generators/GoogleGeneratorTest.php:44]: My company also maintains docs for services like Google calendar: https://github.com/InteractionDesignFoundation/add-event-to-calendar-docs/blob/master/services/google.md

Links to these docs you can find on PHPDoc (class-level docs on generators)

Comment (atymic) [reply tests/Generators/GoogleGeneratorTest.php:44]: Ahh awesome. Didn't realise google support recurring rules. Shame other providers dont

**Pull request merged**